pinot-spi/src/main/java/org/apache/pinot/spi/stream/StreamMetadataProvider.java

```diff
@@ -106,6 +107,33 @@ default Map<String, PartitionLagState> getCurrentPartitionLagState(
     return result;
   }
 
+  /**
+   * Fetches the list of available topics/streams
+   *
+   * @param timeout Timeout for fetching the list of topics. If this is null, the implementation should use a default
+   *                timeout.
+   * @return List of topics
+   */
+  default List<TopicMetadata> listTopics(Duration timeout) {
+    throw new UnsupportedOperationException();
+  }
+
+  /**
+   * Represents the metadata of a topic. This can be used to represent the topic name and other metadata in the future.
+   */
+  class TopicMetadata {
```

Comment:
I think add a map field here to put more topic level information?
Or just return a `List<String>` for `listTopics` call.

Reply:
`List<String>` might be restrictive for further expansion of this API. I'm thinking if we want to expand on partition data etc later on. 

`Map<String, TopicMetadata>` also works. I had `List<>` because the API was called `listTopics()`. wdyt?

Reply:
Or make TopicMetadata an interface then you can have KafkaTopicMetadata